plugins/trace_api_plugin/include/eosio/trace_api/store_provider.hpp

```diff
@@ -184,10 +217,23 @@ namespace eosio::trace_api {
       // take an open index slice file and verify its header is valid and prepare the file to be appended to (or read from)
       void validate_existing_index_slice_file(fc::cfile& index_file, open_state state) const;
 
+      // helper for methods that process irreversible slice files
+      template<typename F>
+      void process_irreversible_slice_range(uint32_t lib, uint32_t upper_bound_block, std::optional<uint32_t>& lower_bound_slice, F&& f);
+
       const boost::filesystem::path _slice_dir;
       const uint32_t _width;
       const std::optional<uint32_t> _minimum_irreversible_history_blocks;
       std::optional<uint32_t> _last_cleaned_up_slice;
+      const std::optional<uint32_t> _minimum_uncompressed_irreversible_history_blocks;
+      std::optional<uint32_t> _last_compressed_slice;
+      uint32_t _compression_seek_points;
+
+      std::atomic<uint32_t> _best_known_lib{0};
+      std::mutex _maintenance_mtx;
+      std::condition_variable _maintenance_condition;
+      std::thread _maintenance_thread;
+      std::atomic_bool _maintenance_shutdown{false};
```

Comment:
This looks fine, but I think it would have been simpler to use `named_thread_pool` with size of 1 and posted jobs using `async_thread_pool`.

Reply:
I'm on the fence here.  If you are fine I will leave it.   The idea of a thread-pool implies that you could have more than one thread in the pool and as this is not safe for parallel background tasks I would want to do a little more work making it resilient to a future engineer making a reasonable assumption that they could parallelize the compression tasks by upping the number of threads in the pool.

Reply:
I'm fine with leaving it as is.